google/cloud/opentelemetry/internal/monitored_resource.cc

```diff
@@ -162,8 +164,8 @@ MonitoredResourceProvider GenericTask() {
           {"location",
            {{sc::kCloudAvailabilityZone, sc::kCloudRegion}, "global"}},
           {"namespace", {{sc::kServiceNamespace}}},
-          {"job", {{sc::kServiceName}}},
-          {"task_id", {{sc::kServiceInstanceId}}},
+          {"job", {{sc::kServiceName, sc::kFaasName}}},
+          {"task_id", {{sc::kServiceInstanceId, sc::kFaasInstance}}},
```

Comment:
Again, good catch and thanks.

Can we have unit tests for these too? Or at least fix the broken tests.

Also, consider leaving a comment in a test case where we fallback to the empty string. Just something that says "Verify that we fallback to the empty string if no matches are found".

Reply:
Yep, I’ll add some tests for these cases.